src/memory/memory.cc

```diff
@@ -1,3 +1,45 @@
 #include "memory/memory.hh"
 
-namespace sim {}
+namespace sim {
+
+Memory::Memory() {}
+
+Word Memory::loadWord(AddrType addr) {
+
+  stats.numLoads++;
+  Word word = kDummyWord;
+  auto it = mem.find(addr);
+  if (it != mem.end())
+    word = it->second;
+  else
+    word = pageFaultHandle(addr);
+  return word;
+}
+
+void Memory::storeWord(AddrType addr, Word word) {
+
+  stats.numStores++;
+  mem[addr] = word;
+}
+
+Word Memory::pageFaultHandle(AddrType addr) {
+
+  stats.numPageFaults++;
+  std::cout << "Placeholder for page fault" << std::endl;
```

Comment:
And here too

Reply:
Yes, but this is a placeholder for the 1st version without physical memory implementation. With the physical memory there in no need of line 28. So there is no need in logging at all.